crates/matrix-sdk-base/src/client.rs

```diff
@@ -610,54 +610,6 @@ impl BaseClient {
         Ok(())
     }
 
-    /// Process the events provided during a sync.
-    ///
-    /// events must be exactly the same list of events that are in raw_events,
-    /// but deserialised. We demand them here to avoid deserialising
-    /// multiple times.
-    #[instrument(skip_all, fields(room_id = ?room_info.room_id))]
-    pub(crate) async fn handle_state(
-        &self,
-        context: &mut Context,
-        raw_events: &[Raw<AnySyncStateEvent>],
-        events: &[AnySyncStateEvent],
-        room_info: &mut RoomInfo,
-        ambiguity_cache: &mut AmbiguityCache,
-    ) -> StoreResult<BTreeSet<OwnedUserId>> {
-        let mut state_events = BTreeMap::new();
-        let mut user_ids = BTreeSet::new();
-
-        assert_eq!(raw_events.len(), events.len());
```

Comment:
Did you deliberately remove this assertion? Isn't it a little helpful?

Reply:
I thought the assertion wasn't particularly helpful. Either it should become a `debug_assert_eq`, or we should introduce a type indicating those two values must be "synced", or to use a tuple as you've noticed.